Add distance measurement tool

spycolyf opened the pull request:
Please review this code. It is my attempt to implement to Distance Measurement Tool. There is a known issue with the edit function that replaces the QREADS tool bar. The absolute best solution would be to allow the user to get the QREADS toolbar back rather than eliminating the very useful Slicer measurement line edit function.

spycolyf committed:
ENH: Support for adding distance measurement tool in slice viewers

Known issues:

* When right mouse click on measurement line and select "Edit"
  in the menu, the QREADS Toolbar goes away and cannot be retrieved.

See #21

Co-authored-by: Jean-Christophe Fillion-Robin <jchris.fillionr@kitware.com>

jcfr commented on Modules/Scripted/QReads/QReads.py:
```diff
@@ -275,6 +276,18 @@ def showSlicerQREADSHelp(self):
       self.helpWebWidget.url = qt.QUrl('http://www.google.com')
     self.helpWebWidget.show()
 
+  def createDistMeasTool(self):
```
No need to abbreviate, consider using fully descriptive name

jcfr commented on Modules/Scripted/QReads/QReads.py:
```diff
@@ -275,6 +276,18 @@ def showSlicerQREADSHelp(self):
+    # Create the SlicerQREADS Orientation Marker Type
+    lineNode = slicer.mrmlScene.CreateNodeByClass("vtkMRMLMarkupsLineNode")
+    lineNode.SetName("Line")
+    slicer.mrmlScene.AddNode(lineNode)
```
```python
lineNode.UnRegister(slicer.mrmlScene)
```

See https://www.slicer.org/wiki/Documentation/Nightly/Developers/Tutorials/MemoryManagement#Python_scripts_and_scripted_modules

jcfr reviewed (changes requested):
Consider rebasing the changes and addressing comments

jcfr committed:
ENH: Remove irrelevant actions when right-clicking on markups

List of Slicer changes:

$ git shortlog 7fde12df2..82b0bce10 --no-merges
Jean-Christophe Fillion-Robin (2):
      [Backport] ENH: Add support for deleting markup using right-click
      [SlicerQReads] ENH: Remove irrelevant actions when right-clicking on markups

See #21

jcfr commented:
Only the relevant actions are visible when right-clicking:

![image](https://user-images.githubusercontent.com/219043/115612561-6578d800-a2b9-11eb-93ab-083928e169d0.png)

spycolyf commented:
I'm currently making the following changes:

1) Make the Distance measurement Tool Button a toggle button.
2) When toggle is checked, enable 1 drawing on any image
3) When toggle is unchecked, Delete line.

jcfr commented:
> I'm currently making the following changes:

Consider creating a new issue and new branch based on the latest version. Thanks

jcfr commented:
> When toggle is unchecked, Delete line.

This has already been implemented using right click.

spycolyf commented:
OK, Could you make the Interactive handles also a a right mouse choice if easy to do

jcfr commented:
> OK, Could you make the Interactive handles also a a right mouse choice if easy to do

Consider creating an issue and associated the "Backlog" milestone. For now, I consider this is a very minor tweak that should be addressed at later stage after allocating resources for this project.

spycolyf commented:
OK
